proof/infoflow/refine/ADT_IF_Refine.thy

```diff
@@ -784,6 +784,17 @@ lemma scheduler_if'_valid_duplicates[wp]:
   apply (wp hoare_drop_imps | simp)+
   done
 
+lemma nonzero_gt_zero[simp]:
+  "x \<noteq> (0 :: obj_ref) \<Longrightarrow> x > 0"
+  apply (rule nonzero_data_to_nat_simp)
+  apply (simp add: unat_gt_0)
+  done
+
+lemma gt_zero_nonzero[simp]:
+  "(0 :: obj_ref) < x \<Longrightarrow> x \<noteq> 0"
```

Comment:
same as for `nonzero_gt_zero`. Maybe declare them as `lemmas` and use them with `notes [simp] = ..` in the lemmas that benefit from them.